File: Sorting Algorithms/Cocktail Sort/CocktailSort.cs
/// <summary>
/// Cocktail Sort Algorithm implementation in C#
/// The algorithm sorts an array of integers by iterating over the array in both directions,
/// swapping elements if they are in the wrong order. The process is repeated until the array is fully sorted.
/// </summary>
/// <param name="array">Array to be sorted</param>
/// <returns>Sorted array</returns>
public static int[] CocktailSort(int[] array)
{
    bool swapped;
    do
    {
        swapped = false;
        for (int i = 0; i <= array.Length - 2; i++)
        {
            if (array[i] > array[i + 1])
            {
                int temp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = temp;
                swapped = true;
            }
        }

        if (!swapped)
        {
            break;
        }

        swapped = false;

        for (int i = array.Length - 2; i >= 0; i--)
        {
            if (array[i] > array[i + 1])
            {
                int temp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = temp;
                swapped = true;
            }
        }
    } while (swapped);

    return array;
}

File: Sorting Algorithms/Cocktail Sort/CocktailSort.java
/**
 * This method implements the Cocktail Sort algorithm to sort an array of integers.
 * 
 * The algorithm sorts an array of integers by iterating over the array in both directions,
 * swapping elements if they are in the wrong order. The process is repeated until the array is fully sorted.
 *
 * @param arr the array to be sorted
 */
public static void cocktailSort(int[] arr) {
    boolean swapped = true;
    int start = 0;
    int end = arr.length;

    while (swapped) {
        swapped = false;

        for (int i = start; i < end - 1; ++i) {
            if (arr[i] > arr[i + 1]) {
                int temp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = temp;
                swapped = true;
            }
        }

        if (!swapped) {
            break;
        }

        swapped = false;
        end--;

        for (int i = end - 1; i >= start; --i) {
            if (arr[i] > arr[i + 1]) {
                int temp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = temp;
                swapped = true;
            }
        }

        start++;
    }
}
